Documentation/hacking.md

```diff
@@ -207,7 +207,47 @@ Now, GOTO [saving](#saving-the-set-of-dependencies)
 
 #### Removing an existing dependency
 
-This is the simplest case of all: simply remove all references to a dependency from the source files.
+This is almost the simplest case of all: simply remove all references to a dependency from the source files.
+
+Now, GOTO [saving](#saving-the-set-of-dependencies)
+
+#### Add a new application
+
+In this case you'll first need to retrieve the application you're working with into `GOPATH`.
+As a simple example, assuming we're adding `github.com/fizz/buzz/app`:
+
+```
+$ go get -d github.com/fizz/buzz
+```
+
+Then add you new application to the vendoredApps file:
+
+```
+$ vim vendoredApps
+...
+github.com/fizz/buzz/app
+...
+```
+
+Now, GOTO [saving](#saving-the-set-of-dependencies)
+
+#### Update an existing application
```

Comment:
Right. Actually, if you are doing only an application update, then even godep save ./... is unnecessary. But let's avoid complicating instructions.

I need to come up with some better layout. How about:

```
#### Add a new dependency

##### If a dependency is a library

Blah blah blah

##### If a dependency is an application

Blah blah blah

#### Update an existing dependency

common Blah blah blah for both application and library

#### Removing an existing dependency

##### If a dependency is a library

Blah blah blah

##### If a dependency is an application

Blah blah blah
```